src/pudl/metadata/resources/mcoe.py

```diff
@@ -0,0 +1,224 @@
+            "etl_group": "outputs",
+        }
+        for freq in AGG_FREQS
+    }
+    | {
+        f"heat_rate_by_generator_{freq}": {
+            "description": (
+                f"{freq.title()} heat rate estimates by generator. These are actually "
+                "just generation unit level heat rates, which have been broadcast "
+                "across all constituent generator IDs, since heat rates really only "
+                "have a well-defined meaning in the context of a generation unit."
+            ),
+            "schema": {
+                "fields": [
+                    "report_date",
+                    "plant_id_eia",
+                    "unit_id_pudl",
+                    "generator_id",
+                    "heat_rate_mmbtu_mwh",
+                    "fuel_type_code_pudl",
+                    "fuel_type_count",
+                ],
+                "primary_key": [
+                    "report_date",
+                    "plant_id_eia",
+                    "generator_id",
+                ],
+            },
+            "field_namespace": "eia",
+            "sources": ["eia923", "eia860"],
+            "etl_group": "outputs",
+        }
+        for freq in AGG_FREQS
+    }
+    | {
+        f"capacity_factor_by_generator_{freq}": {
+            "description": (
+                f"{freq.title()} estimates of generator capacity factor. Capacity "
+                "factor is calculated based on reported generator capacity and the "
+                "allocated net generation reported in the generation and generation "
+                "fuel tables."
+            ),
+            "schema": {
+                "fields": [
+                    "report_date",
+                    "plant_id_eia",
+                    "generator_id",
+                    "net_generation_mwh",
+                    "capacity_mw",
+                    "capacity_factor",
+                ],
+                "primary_key": [
+                    "report_date",
+                    "plant_id_eia",
+                    "generator_id",
+                ],
+            },
+            "field_namespace": "eia",
+            "sources": ["eia923", "eia860"],
+            "etl_group": "outputs",
+        }
+        for freq in AGG_FREQS
+    }
+    | {
+        f"fuel_cost_by_generator_{freq}": {
+            "description": (
+                f"{freq.title()} estimate of per-generator fuel costs both per MMBTU "
+                "and per MWh. These calculations are based on the allocation of net "
+                "generation and plant-level delivered fuel prices reported in the fuel "
+                "receipts and cost table, along with the un-allocated fuel consumption "
+                "from the boiler fuel table. This means fuel cost coverage is low."
+                "The fuel costs are also currently aggregated to coarse fuel "
+                "categories rather than using the more detailed energy source codes."
```

Comment:
If the heat rates are based on allocated fuel consumption, why should the fuel costs not also be based on the allocated fuel consumption?  Is this documentation from when we had done the net generation allocation, but hadn't yet allocated the fuel consumption?

What fuel cost coverage do we expect here, in terms of overall records, and in terms of fraction of total generation? Why are the fuel costs aggregated to coarse categories rather than the energy source codes?

Should investigate, explain, and caveat negative or other ridiculous fuel costs per MWh.